services/cache/lru.go

```diff
@@ -226,23 +235,6 @@ func (l *LRU) get(key string, value interface{}) error {
 	return msgpack.Unmarshal(e.value, value)
 }
 
-func (l *LRU) getItem(key string) (*entry, error) {
```

Comment:
From what I can see the race is just about accessing the entry struct, not about the byte slice. We can guarantee the data is only written before any read so it shouldn't be a problem. Wouldn't returning `e.value` here instead of the whole struct be sufficient to fix this case?

Reply:
Yep, indeed. My bad.

Reply:
Just out of curiosity, where does data race message tell you if race was for struct and not byte slice?

I am referring to this
https://mattermost.atlassian.net/browse/MM-30863

Reply:
It does not directly tell you. Needs a little bit of fiddling to understand it.